centrallix-sysdoc/OSDriver_Overview.md

```diff
@@ -0,0 +1,296 @@
+
+    1.	Centrallix --> OSD		Initialize() routine called.
+    2.	               OSD		Driver initializes itself
+    3.	               OSD --> OSML	Driver registers with OSML
+    4.	               OSD <-- OSML	Register succeeds or fails
+    5.	Centrallix <-- OSD		Init succeeds or fails
+
+### B.	Accessing Objects.
+When the OSML needs to access an object handled by the driver, then the driver's Open() routine is called.  The result of the Open will be a handle (opaque pointer) to the open object from the driver's perspective.  When the OSML is done with the object, it will call the Close() routine for that object, passing the same opaque pointer to the driver's Close() routine that was obtained from the Open() routine.
+
+The driver will need to access the node object via the OSML to figure out where to connect to externally or to obtain the content that it should be breaking down into meaningful structure.
+
+    1.	OSML --> OSD			Open object
+    2.	         OSD --> OSML		Accesses node using OSML API
+    3.	         OSD <-- OSML		Driver gets data for node obj.
+    4.	OSML <-- OSD			Returns object handle
+
+    5.	OSML --> OSD			Close object
+    6.	         OSD			Cleans up object
+    7.	OSML <-- OSD			Close routine returns
+
+Here is a more detailed look at how a row in our CSV file from the previous example might be opened.  Note that many of the details have been left out in order to highlight the overall process.
+
+In this view, the various pieces are abbreviated as follows to make things fit on the page:  U=user, O=OSML, R=rootnode driver, F=filesystem driver, D=datafile driver.
+
+    1.	U -> O			User requests that the object /DirectoryOne/FileTwo.csv/rows/0 be opened.
+    2.	     O -> R		Open an instance of the / object and
+            O <- R		use that as the first object in the 
+                chain of open objects needed to get 
+                at the object the user wants.
+    3.	     O			Discovers that rootnode points to a local filesystem tree.
+    4.	     O -> F		OSML tries to get local filesystem driver to open the given path, /DirectoryOne/FileTwo.csv/rows/0
+    5.	     O <- F		Filesystem driver reports that it can only open /DirectoryOne/FileTwo.csv, and returns a handle for it.
+    6.	     O			Adds an object to the chain of open objects for the open file, and figures out that the content of the FileTwo.csv is CSV data.
+    7.	     O -> D		OSML tries to get the datafile driver to open up FileTwo.csv/rows/0, the remainder of the path.
+    8.	          D -> O	datafile driver uses the OSML API to access the open object created in step 5 above (the file).
+    9.	               O -> F	OSML requests content for the file from the local filesystem driver.
+    1.                 O <- F	Local filesystem driver returns the needed content.
```

Comment:
Little oddity here -- the numbering has 1. instead of 10. and the O <- F isn't indented right.